test-documentation.md

```diff
@@ -0,0 +1,1128 @@
+each of them can report a unique hostname.
+
+
+
+## [pods-created-pod-assigned-hostip](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/pods.go#L136)
+
+Make sure when a pod is created that it is assigned a host IP
+Address.
+
+
+
+## [pods-submitted-removed](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/pods.go#L158)
+
+Makes sure a pod is created, a watch can be setup for the pod,
+pod creation was observed, pod is deleted, and pod deletion is observed.
+
+
+
+## [pods-updated-successfully](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/pods.go#L283)
+
+Make sure it is possible to successfully update a pod's labels.
+
+
+
+## [pods-update-active-deadline-seconds](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/pods.go#L338)
+
+Make sure it is possible to create a pod, update its
+activeDeadlineSecondsValue, and then waits for the deadline to pass
+and verifies the pod is terminated.
+
+
+
+## [pods-contain-services-environment-variables](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/pods.go#L384)
+
+Make sure that when a pod is created it contains environment
+variables for each active service.
+
+
+
+## [projected-secret-no-defaultMode](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/projected.go#L42)
+
+Simple projected Secret test with no defaultMode set.
+
+
+
+## [projected-secret-with-defaultMode](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/projected.go#L50)
+
+Simple projected Secret test with defaultMode set.
+
+
+
+## [projected-secret-with-nonroot-defaultMode-fsGroup](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/projected.go#L60)
+
+Simple projected Secret test as non-root with
+defaultMode and fsGroup set.
+
+
+
+## [projected-secret-simple-mapped](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/projected.go#L73)
+
+Simple projected Secret test, by setting a secret and
+mounting it to a volume with a custom path (mapping) on the pod with
+no other settings and make sure the pod actually consumes it.
+
+
+
+## [projected-secret-with-item-mode-mapped](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/projected.go#L82)
+
+Repeat the projected-secret-simple-mapped but this time
+with an item mode (e.g. 0400) for the secret map item.
+
+
+
+## [projected-secret-multiple-volumes](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/projected.go#L113)
+
+Make sure secrets works when mounted as two different
+volumes on the same node.
+
+
+
+## [projected-secret-simple-optional](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/projected.go#L206)
+
+Make sure secrets works when optional updates included.
+
+
+
+## [projected-volume-configMap-nomappings-succeeds](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/common/projected.go#L408)
+
+Part 2/3 - ConfigMaps
```

Comment:
where are the other parts?  Perhaps we should remove (or move) this comment?

Reply:
Not part of this tool, must be addressed separately. This is the other purpose of this document that we identify the fixes.